### PR TITLE
Add pub.dev/topics page

### PR DESCRIPTION
**Pull request by szakarias**

### CODE REVIEW

**Comment by szakarias** [comment]
image

File: app/lib/frontend/handlers/misc.dart [hunk 1]
```diff
@@ -67,6 +73,25 @@ Future<shelf.Response> readinessCheckHandler(shelf.Request request) async {
...
+        .readAsBytes(topicsJsonFileName);
+
+    topics = utf8JsonDecoder.convert(data) as Map<String, int>;
+  } on FormatException {
+    topics = {};
+  } on DetailedApiRequestError catch (e, st) {
+    if (e.status == 404) {
+      topics = {};
+    }
+  }
```
Comment (jonasfj) [inline app/lib/frontend/handlers/misc.dart:91]:
```suggestion
  } on FormatException {
    topics = {};
    // TODO: Log error loading topics
  } on DetailedApiRequestError catch (e, st) {
    topics = {};
    if (e.status == 404) {
      // TODO: Log inability to load topics
    }
  }
```

**Review by jonasfj — APPROVED** [review]

File: app/lib/frontend/handlers/misc.dart (hunk 1 above)
Comment (jonasfj) [reply app/lib/frontend/handlers/misc.dart:91]: We can do the TODO later, I don't think that's particularly blocking.

If we want it we just need:
```dart
import 'package:logging/logging.dart' show Logger;

final _log = Logger('pub.handlers.misc');
```
at the top of the file.

Comment (szakarias) [reply app/lib/frontend/handlers/misc.dart]: added loggin.

File: app/lib/frontend/templates/views/page/topics_list.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+d.Node renderTopicsList(Map<String, int> topics) {
+  return d.div(children: [
+    d.h1(text: 'Topics '),
+    topicsListNode(topics),
+  ]);
+}
+
+/// Renders the package list node of /topics page.
+d.Node topicsListNode(Map<String, int> topics) {
+  final sortedTopics = SplayTreeMap<String, int>.from(
```
Comment (isoos) [inline app/lib/frontend/templates/views/page/topics_list.dart:20]: Isn't this sorted in the file that we store in the bucket?

Comment (isoos) [inline app/lib/frontend/templates/views/page/topics_list.dart:20]: `topics.entries.toList()..sort((a, b) => -a.value.compareTo(b.value))` may be also good here, not sure which consumes less CPU

Comment (szakarias) [reply app/lib/frontend/templates/views/page/topics_list.dart:20]: No.

Comment (szakarias) [reply app/lib/frontend/templates/views/page/topics_list.dart:20]: Let's do this. It's also easier to read.

**Review by isoos — APPROVED** [review]

**Pull request merged**